### PR TITLE
Port Docs to GitBook, Reformat to take advantage of GitBook features, and do some general tidying up.

### PR DESCRIPTION
**Pull request by nmarwell26**
This PR aims to set the stage for more legible documentation. It does this in the following ways:
1) Transitions abstractions.py to markdown format (now abstractions.md) to make it far easier to follow. Note that there are a lot of additional updates to be made to abstractions.py. This also sets the stage for those.
2) Splits core concepts from supplementary ones (such as splitting the global and file-specific environment variables into 2 tables).
3) Fixes grammar and typos.
4) Adds a few helpful tips that were previously missing, such as GraphViz Installation guide for MacOS.

There is a long way to go for tinygrad docs, but I see this as a useful first step in that journey.

### CODE REVIEW

**Comment by nmarwell26** [comment]
Failing tests because abstractions.py became abstractions.md. Abstractions.md meaningfully more legible IMO. Wondering if we should maintain both a .py and a .md version or if it would be best to simply remove that test.

**Comment by wozeparrot** [comment]
you should be able to do what is done for quickstart.md and test the md directly

File: docs/adding_new_accelerators.md [hunk 1]
```diff
@@ -1,10 +1,11 @@
-# Adding a new accelerator to tinygrad
+# Adding New Accelerators
 
 It's pretty easy to add a new accelerator to tinygrad. All you need to do is implement a total of 26 (optionally 27) low level ops. Then tinygrad takes care of the rest, handling derivatives and syntactic sugar.
 
 ## llops
 
-These are the ops that you must implement for your accelerator of choice. Compiled Accelerators do not need to implement movement_ops, as they are handled by the ShapeTracker.
+These are the ops that you must implement for your accelerator of choice. Compiled Accelerators do not need to implement movement\_ops, as they are handled by the ShapeTracker.
```
Comment (wozeparrot) [inline docs/adding_new_accelerators.md:7]: extra capitalization on Accelerators

File: docs/environment_variables/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+* Variable
+* Possible Value(s)
+* Description.
+
+A `#` means that the variable can take any integer value.
+
+### Global Environment Variables
+
+Global Environment Variables control core tinygrad behavior and are found [here](/docs/environment_variables/global_variables.md).
+
```
Comment (wozeparrot) [inline docs/environment_variables/README.md:18]: extra space?

Comment (nmarwell26) [reply docs/environment_variables/README.md:18]: Fixed. Thanks for the catch.

File: docs/adding_new_accelerators.md (hunk 1 above)
Comment (nmarwell26) [reply docs/adding_new_accelerators.md]: Fixed. Thanks for the catch.

**Comment by wozeparrot** [comment]
Don't really like the merge multiple lines into one, it will make reviewing changes in the future harder and it just makes the line really long.

**Comment by nmarwell26** [comment]
> Don't really like the merge multiple lines into one, it will make reviewing changes in the future harder and it just makes the line really long.

Is this in reference to a specific line? @wozeparrot

**Comment by geohot** [comment]
This is way too long to review. Please submit what's valuable in broken up PRs.

**Pull request closed without merging**